tests/test/wf_lowering.rs

```diff
@@ -1103,3 +1103,17 @@ fn no_unsize_impls() {
         }
     }
 }
+
+#[test]
+fn ill_formed_opaque_ty() {
+    lowering_error! {
+        program {
+            trait Foo {}
+            struct Bar {}
```

Comment:
try this test

```rust
trait Foo { }
struct NotFoo { }
struct IsFoo { }
impl Foo for IsFoo { }
opaque type T: Foo = NotFoo;
```

I think your code will accept it, but I could be wrong

Reply:
It does reject this actually, since you still need to prove `NotFoo: Foo`. But, regardless, your point still stands